### PR TITLE
reduce server-side fs reads, refs #332

### PR DESCRIPTION
**Pull request by gl2748**
closes https://github.com/steemit/steempunks/issues/332

With the changes in this PR and following profiling commands

I am seeing:
```
232.76 kb/s avg before.
253.1 kb/s avg after.

8.73 % increase in kb/s.

685.3 req/100s before
899 req/100s after

31.24% increase in requests/s
```

Per:
https://gist.github.com/gl2748/5ee5b145ef2f168895cce74a3466f321

This on the parent process as reported by:
```
yarn run build && SDC_DATABASE_URL="mysql://root@localhost/steemit_dev" NODE_ENV=production node --prof --log_timer_events lib/server/index.js
```
So not too bad. I have a hunch that fs reads for translations might also be slowing things down server side.

### See below.

### CODE REVIEW

File: src/server/app_render.jsx [hunk 1]
```diff
@@ -88,20 +74,11 @@ async function appRender(ctx) {
...
-        const assets_filename =
-            ROOT +
-            (process.env.NODE_ENV === 'production'
-                ? '/tmp/webpack-stats-prod.json'
-                : '/tmp/webpack-stats-dev.json');
-        const assets = require(assets_filename);
-
         // Don't cache assets name on dev
+        // TODO: Investigate removing this.
         if (process.env.NODE_ENV === 'development') {
```
Comment (bnchdrff) [inline src/server/app_render.jsx]: what happens when `process.env.NODE_ENV === 'development'`? it seems like the assets data required at startup will be removed but it will not be restored.

[the remark this answers is not in the document]
Comment (gl2748) [reply src/server/app_render.jsx]: I think originally it might be to get some-kind of hot-reloading of assets for local dev. I think it can be removed as we do not typically change the assets while developing locally and requiring a server-restart is not as onerous as perhaps it once was...

https://nodejs.org/api/modules.html#modules_caching

File: src/server/server.js [hunk 2]
```diff
@@ -78,6 +78,15 @@ if (env === 'development') {
     );
 }
 
+if (process.env.NODE_ENV === 'production') {
+    const resolvedAssets = require(path.join(
```
Comment (bnchdrff) [inline src/server/server.js]: define resolvedAssets outside of this conditional block so that it's false rather than undefined in the surrounding block where it's used later, to reduce ambiguity

File: webpack/base.config.js [hunk 3]
```diff
@@ -59,6 +59,7 @@ module.exports = {
             {test: /\.(jpe?g|png)/, use: 'url-loader?limit=4096'},
             {test: /\.json$/, use: 'json-loader'},
             {test: /\.js$|\.jsx$/, exclude: /node_modules/, use: 'babel-loader'},
+            // All svgs are loaded inline...
```
Comment (bnchdrff) [inline webpack/base.config.js]: put this in another pr where we address the removal of svgs from the webpack stats output

File: src/server/server.js [hunk 4]
```diff
@@ -78,6 +78,15 @@ if (env === 'development') {
...
+        __dirname,
+        '../..',
+        '/tmp/webpack-stats-prod.json'
+    ));
+    const supportedLocales = getSupportedLocales();
```
Comment (bnchdrff) [inline src/server/server.js]: same thing here -- if supportedLocales is defined as false and then conditionally set to the output of getSupportedLocales then we can scrap the conditional mess below

File: src/server/app_render.jsx [hunk 5]
```diff
@@ -88,20 +75,14 @@ async function appRender(ctx) {
             userPreferences,
             offchain,
         });
-
-        // Assets name are found in `webpack-stats` file
```
Comment (bnchdrff) [inline src/server/app_render.jsx]: please keep these & other comments

Comment (gl2748) [reply src/server/app_render.jsx]: Done.

File: src/server/app_render.jsx [hunk 6]
```diff
@@ -88,20 +75,15 @@ async function appRender(ctx) {
...
-                ? '/tmp/webpack-stats-prod.json'
-                : '/tmp/webpack-stats-dev.json');
-        const assets = require(assets_filename);
-
-        // Don't cache assets name on dev
-        if (process.env.NODE_ENV === 'development') {
+        let assets;
+        if (!resolvedAssets) {
+            // Assets name are found in `webpack-stats` file
+            const assets_filename = ROOT + '/tmp/webpack-stats-dev.json';
```
Comment (roadscape) [inline src/server/app_render.jsx]: Don't we need to keep the `process.env.NODE_ENV === 'production'` switch?

Comment (roadscape) [reply src/server/app_render.jsx]: Oh, I see. `webpack-stats-prod.json` is resolved in `src/server/server.js` for `production`.

I guess the dev logic is here because it allows hot reloading?

**Comment by roadscape** [comment]
Changes LGTM, ready for testing
